### PR TITLE
[Feature/extensions] Adding extension framework support for first extension point 

### PR DESCRIPTION
**Pull request by saratvemulapalli**
### Description
Adding support for first example extension point and some refactoring.
This change is for feature/extensions branch.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    private void extensionsDiscovery() throws IOException {
+        logger.info("Extensions Config Directory :" + extensionsPath.toString());
+        if (!FileSystemUtils.isAccessibleDirectory(extensionsPath, logger)) {
+            return;
+        }
+        for (final Path plugin : PluginsService.findPluginDirs(extensionsPath)) {
+            try {
+                PluginInfo pluginInfo = PluginInfo.readFromProperties(plugin);
+                /*
+                 * TODO: Read from extensions.yml
```
Comment (owaiskazi19) [inline server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java:93]: nit: Can we add the issue here?

File: server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+        try {
+            logger.info("Sending request to extension");
+            transportService.sendRequest(
+                extensionNode,
+                INDICES_EXTENSION_POINT_ACTION_NAME,
+                new IndicesModuleRequest(indexModule),
+                indicesModuleResponseHandler
+            );
+            inProgressLatch.await(100, TimeUnit.SECONDS);
```
Comment (owaiskazi19) [inline server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java:242]: nit: Can we add a comment here to mention the async call?

File: server/src/main/java/org/opensearch/index/IndicesModuleResponse.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
```
Comment (owaiskazi19) [inline server/src/main/java/org/opensearch/index/IndicesModuleResponse.java:10]: Do we need this license here?

**Review by owaiskazi19 — APPROVED** [review]
Overall LGTM! Added few nitpicks

File: server/src/main/java/org/opensearch/index/IndicesModuleResponse.java (hunk 3 above)
Comment (saratvemulapalli) [reply server/src/main/java/org/opensearch/index/IndicesModuleResponse.java]: Thats a great catch. Fixed!

**Pull request merged**